Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,355 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        const string InValidOutput = "N/A."; 
+        public static bool IsInValidValidInput(string[] input) => input == null || input.Length == 0;
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var HighestBalanceEver = decimal.MinValue;
+            var PersonWithHighestBalance = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                HighestBalanceEver = FindHighestBalance(peopleAndBalances, HighestBalanceEver, PersonWithHighestBalance, i);
+
+            }
+
+            var answer = $"{PersonWithHighestBalance.ToString()} had the most money ever. ¤{HighestBalanceEver}.";
+
+            return answer;
+        }
+
+        private static decimal FindHighestBalance(string[] peopleAndBalances, decimal HighestBalanceEver, StringBuilder PersonWithHighestBalance, int i)
+        {
+            var currentPersonData = peopleAndBalances[i].Split(',');
+            var highestAmountOfPerson = FindHighestBalanceOfPerson(currentPersonData);
+
+            if (highestAmountOfPerson > HighestBalanceEver)
+            {
+                HighestBalanceEver = highestAmountOfPerson;
+                var currentOne = PersonWithHighestBalance.ToString();
+
+                if (string.IsNullOrEmpty(currentOne))
+                {
+                    PersonWithHighestBalance.Append(currentPersonData[0]);
+                }
+                else
+                {
+                    PersonWithHighestBalance.Clear();
+                    PersonWithHighestBalance.Append(currentPersonData[0]);
+                }
+
+
+            }
+            else if (highestAmountOfPerson == HighestBalanceEver)
+            {
+                if (i == peopleAndBalances.Length - 1)
+                {
+                    PersonWithHighestBalance.Append(" and ");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Define a constant instead of using this literal ' and ' 4 times.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4937470)